src/lazy/expanded/compiler.rs

```diff
@@ -450,12 +455,11 @@ impl TemplateCompiler {
     /// [`TemplateBodyExpr`] sequences to the `TemplateBody`.
     ///
     /// If `is_literal` is true, nested symbols and s-expressions will not be interpreted.
-    fn compile_value<'top, D: Decoder>(
-        context: EncodingContextRef<'top>,
-        signature: &MacroSignature,
+    fn compile_value<D: Decoder>(
+        tdl_context: TdlContext,
```

Comment:
🗺️ Rather than passing a `context`, `signature`, and (now) `pending_macros` to the compiler for each subexpression, we now pass a single `TdlContext` (defined below on line 840). We can extend this `TdlContext` as needed in the future to avoid similar plumb-the-argument-through PRs.